workers/handshake-manager/src/manager/internal_engine.rs

```diff
@@ -62,22 +62,33 @@ impl HandshakeExecutor {
 
```

Comment:
might make more sense to move this whole check into a pre-filter implemented in `get_locally_matchable_orders`. We could change that method into `get_locally_matchable_orders_in_pool` or something of the like. This scales a bit better as we don't need to open and close a tx for each new order and all the filtering can be done up front. Plus putting the check where it is now, we incur a bunch of other lookups ahead of even getting to the order (e.g. looking up the wallets for both orders).

This is why, for example, the implementation of `get_locally_matchable_orders` does all these checks up front; ensuring there's a wallet, ensuring proofs exist, making sure the queue isn't paused, etc. I feel that's a more natural home for this sort of filtering.

Reply:
yeah i totally see where you're coming from. my initial approach was to do this in `get_locally_matchable_orders` but then I saw it's used in a number of other places, e.g. the open orders admin endpoint & the handshake manager, and I feel in those places filtering by matching pool does not make sense. but I like the approach of making a separate `get_locally_matchable_orders_in_pool`, I'll go ahead with that.